packages/core/src/common/keybinding.ts

```diff
@@ -137,45 +137,66 @@ export class KeybindingRegistry {
         bindings.push(binding);
         this.keybindings[keyCode.keystroke] = bindings;
 
-        const commands = this.commands[commandId] || [];
-        commands.push(binding);
-        this.commands[commandId] = bindings;
+        /* Keep a mapping of the Command -> Key mapping.  */
+        if (!this.isPseudoCommand(commandId)) {
+            const commands = this.commands[commandId] || [];
+            commands.push(binding);
+            this.commands[commandId] = bindings;
+        }
     }
 
     /**
-     * The `active` flag with `false` could come handy when we do not want to check whether the command is currently active or not.
-     * For instance, when building the main menu, it could easily happen that the command is not yet active (no active editors and so on)
-     * but still, we have to build the key accelerator.
+     * Get the keybindings associated to commandId.
      *
-     * @param commandId the unique ID of the command for we the associated ke binding are looking for.
-     * @param options if `active` is false` then the availability of the command will not be checked. Default is `true`
+     * @param commandId The ID of the command for which we are looking for keybindings.
      */
-    getKeybindingForCommand(commandId: string, options: { active: boolean } = ({ active: true })): Keybinding | undefined {
-        const bindings = this.commands[commandId];
-        if (!bindings) {
-            return undefined;
-        }
-        if (!options.active) {
-            return bindings[0];
-        }
-        return bindings.find(this.isActive.bind(this));
+    getKeybindingsForCommand(commandId: string): Keybinding[] {
+        return this.commands[commandId] || [];
```

Comment:
Ideally, this should also respect the context. I.e. if you open command palette from within terminal it should not show the disabled keybindings. I think we can do that in a subsequent PR, though.

Reply:
Ok.